File: ANTERIOR/Projeto base - apCaminhosMarte/apCaminhosMarte/Caminho.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

public class Caminho : IComparable<Caminho>, ICopiavel<Caminho>
{
    private int idCidadeOrigem, idCidadeDestino, distancia, tempo, custo;
    const int inicioIdCidadeOrigem = 0;
    const int tamanhoIdCidadeOrigem = 3;
    const int inicioIdCidadeDestino = inicioIdCidadeOrigem + tamanhoIdCidadeOrigem;
    const int tamanhoIdCidadeDestino = 3;
    const int inicioDistancia = inicioIdCidadeDestino + tamanhoIdCidadeDestino;
    const int tamanhoDistancia = 5;
    const int inicioTempo = inicioDistancia + tamanhoDistancia;
    const int tamanhoTempo = 4;
    const int inicioCusto = inicioTempo + tamanhoTempo;

    public int IdCidadeOrigem { get => idCidadeOrigem; set => idCidadeOrigem = value; }
    public int IdCidadeDestino { get => idCidadeDestino; set => idCidadeDestino = value; }
    public int Distancia { get => distancia; set => distancia = value; }
    public int Tempo { get => tempo; set => tempo = value; }
    public int Custo { get => custo; set => custo = value; }
    

    public Caminho(string linha)
    {
        this.IdCidadeOrigem = int.Parse(linha.Substring(inicioIdCidadeOrigem,tamanhoIdCidadeOrigem));
        this.IdCidadeDestino = int.Parse(linha.Substring(inicioIdCidadeDestino,tamanhoIdCidadeDestino));
        this.Distancia = int.Parse(linha.Substring(inicioDistancia,tamanhoDistancia));
        this.Tempo = int.Parse(linha.Substring(inicioTempo,tamanhoTempo));
        this.Custo = int.Parse(linha.Substring(inicioCusto));
    }

    public Caminho()
    {
        this.IdCidadeOrigem = default(int);
        this.IdCidadeDestino = default(int);
        this.Distancia = default(int);
        this.Tempo = default(int);
        this.Custo = default(int);
    }

    public int CompareTo(Caminho outro)
    {
        return this.CompareTo(outro);
    }

    public Caminho Copia()
    {
        Caminho c = new Caminho();
        c.IdCidadeOrigem = IdCidadeOrigem;
        c.IdCidadeDestino = IdCidadeDestino;
        c.Distancia = Distancia;
        c.Tempo = Tempo;
        c.Custo = Custo;

        return c;
    }
}

File: ProjetoAndroid/app/src/main/java/br/unicamp/cotuca/pathbetweencities/caminho/Caminho.java
package br.unicamp.cotuca.pathbetweencities.caminho;

public class Caminho implements Comparable<Caminho> {
    private int idCidadeOrigem, idCidadeDestino, distancia, tempo, custo;

    public int getIdCidadeOrigem() {
        return idCidadeOrigem;
    }

    public void setIdCidadeOrigem(int idCidadeOrigem) {
        this.idCidadeOrigem = idCidadeOrigem;
    }

    public int getIdCidadeDestino() {
        return idCidadeDestino;
    }

    public void setIdCidadeDestino(int idCidadeDestino) {
        this.idCidadeDestino = idCidadeDestino;
    }

    public int getDistancia() {
        return distancia;
    }

    public void setDistancia(int distancia) {
        this.distancia = distancia;
    }

    public int getTempo() {
        return tempo;
    }

    public void setTempo(int tempo) {
        this.tempo = tempo;
    }

    public int getCusto() {
        return custo;
    }

    public void setCusto(int custo) {
        this.custo = custo;
    }

    final int inicioIdCidadeOrigem = 0;
    final int tamanhoIdCidadeOrigem = 3;
    final int inicioIdCidadeDestino = inicioIdCidadeOrigem + tamanhoIdCidadeOrigem;
    final int tamanhoIdCidadeDestino = 3;
    final int inicioDistancia = inicioIdCidadeDestino + tamanhoIdCidadeDestino;
    final int tamanhoDistancia = 5;
    final int inicioTempo = inicioDistancia + tamanhoDistancia;
    final int tamanhoTempo = 4;
    final int inicioCusto = inicioTempo + tamanhoTempo;


    public Caminho(String linha)
    {
        this.setIdCidadeOrigem(Integer.parseInt(linha.substring(inicioIdCidadeOrigem,tamanhoIdCidadeOrigem)));
        this.setIdCidadeDestino(Integer.parseInt(linha.substring(inicioIdCidadeDestino,tamanhoIdCidadeDestino)));
        this.setDistancia(Integer.parseInt(linha.substring(inicioDistancia,tamanhoDistancia)));;
        this.setTempo(Integer.parseInt(linha.substring(inicioTempo,tamanhoTempo)));
        this.setCusto(Integer.parseInt(linha.substring(inicioCusto)));
    }

    public Caminho()
    {
        this.setIdCidadeOrigem(-1);
        this.setIdCidadeDestino(-1);
        this.setDistancia(-1);
        this.setTempo(-1);
        this.setCusto(-1);
    }

    @Override
    public int compareTo(Caminho o) {
        return 0;
    }
}
